src/Extension/SecurityAdminAccountResetExtension.php

```diff
@@ -0,0 +1,156 @@
+ * @property SecurityAdmin $owner
+ */
+class SecurityAdminAccountResetExtension extends Extension
+{
+    use JSONResponse;
+
+    private static $allowed_actions = [
+        'reset',
+    ];
+
+    /**
+     * @var string[]
+     */
+    private static $dependencies = [
+        'Logger' => '%$' . LoggerInterface::class . '.account_reset',
+    ];
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    public function reset(HTTPRequest $request): HTTPResponse
+    {
+        if (!$request->isPOST() || !$request->param('ID')) {
+            return $this->jsonResponse(
+                [
+                    'error' => _t(__CLASS__ . '.BAD_REQUEST', 'Invalid request')
+                ],
+                400
+            );
+        }
+
+        $body = json_decode($request->getBody() ?? '', true);
+
+        if (!SecurityToken::inst()->check($body['csrf_token'] ?? null)) {
+            return $this->jsonResponse(
+                [
+                    'error' => _t(__CLASS__ . '.INVALID_CSRF_TOKEN', 'Invalid or missing CSRF token')
+                ],
+                400
+            );
+        }
+
+        if (!Permission::check(MemberExtension::MFA_ADMINISTER_REGISTERED_METHODS)) {
+            return $this->jsonResponse(
+                [
+                    'error' => _t(
+                        __CLASS__ . '.INSUFFICIENT_PERMISSIONS',
+                        'Insufficient permissions to reset user'
+                    )
+                ],
+                403
+            );
+        }
+
+        /** @var Member $memberToReset */
+        $memberToReset = Member::get()->byID($request->param('ID'));
+
+        if ($memberToReset === null) {
+            return $this->jsonResponse(
+                [
+                    'error' => _t(
+                        __CLASS__ . '.INVALID_MEMBER',
+                        'Requested member for reset not found'
+                    )
+                ],
+                403
+            );
+        }
+
+        $sent = $this->sendResetEmail($memberToReset);
+
+        if (!$sent) {
+            return $this->jsonResponse(
+                [
+                    'error' => _t(
+                        __CLASS__ . '.EMAIL_NOT_SENT',
+                        'Email sending failed'
+                    )
+                ],
+                500
+            );
+        }
+
+        return $this->jsonResponse(['success' => true], 200);
+    }
+
+    /**
+     * @param Member&MemberResetExtension $member
+     * @return bool
+     * @throws ValidationException
+     * @throws PasswordEncryptor_NotFoundException
+     */
+    protected function sendResetEmail($member)
+    {
+        // Generate / store / obtain reset token
+        $token = $member->generateAccountResetTokenAndStoreHash();
+
+        // Create email and fire
+        try {
+            $email = Email::create()
+                ->setHTMLTemplate('SilverStripe\\MFA\\Email\\AccountReset')
```

Comment:
We just merged #134 which contains some templates. This looks like a good file path, would you mind updating the others to match?

Reply:
Done!